{
  try {
    if (iterator != null)     iterator.close();
  }
 catch (  Exception e) {
    getLogger().error("Error in closing iterator",e);
  }
}
